src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c

```diff
@@ -1437,7 +1437,9 @@ static int xdma_process_requests(struct xdma_engine *engine,
 		timeout = wait_event_interruptible_timeout(req->arbtr_wait,
 				req->sw_desc_cnt == req->desc_completed,
 				msecs_to_jiffies(10000)); /* 10sec timeout */
-		if (timeout == 0) {
+
+		/* Check for timeouts or errors */
+		if (timeout == 0 || timeout < 0) {
```

Comment:
Could we print different dmesg for ctrl-c case?

Reply:
Yes I can make that change. can we say "Request aborted with error code = "

Reply:
Sounds great to me.